en/examples/transition_vtol_fixed_wing.md

```diff
@@ -0,0 +1,242 @@
+
+The example terminal output for a debug build of DroneCore should be similar to that shown below (a release build will omit the "Debug" messages):
+
+```
+$ ./transition_vtol_fixed_wing 
+Waiting to discover device...
+[10:24:42|Info ] New device on: 127.0.0.1:14557 (udp_connection.cpp:210)
+[10:24:42|Debug] MAVLink: info: [logger] file: rootfs/fs/microsd/log/2017-11-21/0 (device_impl.cpp:229)
+[10:24:42|Debug] MAVLink: info: Landing detected (device_impl.cpp:229)
+[10:24:43|Debug] Discovered 4294967298 (dronecore_impl.cpp:234)
+Discovered device with UUID: 4294967298
+Arming...
+Taking off...
+[10:24:44|Debug] MAVLink: info: ARMED by arm/disarm component command (device_impl.cpp:229)
+[10:24:44|Debug] MAVLink: info: Using minimum takeoff altitude: 10.00 m (device_impl.cpp:229)
+[10:24:44|Debug] MAVLink: info: Takeoff detected (device_impl.cpp:229)
+[10:24:44|Debug] MAVLink: critical: Using minimum takeoff altitude: 10.00 m (device_impl.cpp:229)
+Altitude: 0.079 m
+Altitude: 0.507 m
+...
+Altitude: 10.254 m
+Transition to fixedwing...
+Altitude: 10.263 m
+...
+Altitude: 20.72 m
+Altitude: 24.616 m
+Altitude: 22.262 m
+Transition back to multicopter...
+Altitude: 17.083 m
+...
+Return to launch...
+Altitude: 11.884 m
+[10:25:09|Debug] MAVLink: info: RTL: climb to 518 m (29 m above home) (device_impl.cpp:229)
+Altitude: 13.61 m
+...
+Altitude: 27.489 m
+Altitude: 28.892 m
+[10:25:18|Debug] MAVLink: info: RTL: return at 517 m (29 m above home) (device_impl.cpp:229)
+Altitude: 29.326 m
+Altitude: 29.33 m
+...
+Altitude: 29.323 m
+Altitude: 29.357 m
+Landing...
+[10:25:29|Debug] MAVLink: info: Landing at current position (device_impl.cpp:229)
+Altitude: 29.199 m
+Altitude: 28.722 m
+Altitude: 28.189 m
+Altitude: 27.62 m
+Finished...
+```
+
+## How it works
+
+The operation of most of this code is discussed in the guide: [Takeoff and Landing (and other actions)](../guide/taking_off_landing.md#transition_vtol).
+
+## Source code {#source_code}
+
+> **Tip** The full source code for the example [can be found on Github here](https://github.com/dronecore/DroneCore/tree/{{ book.github_branch }}/example/transition_vtol_fixed_wing).
+
+
+[CMakeLists.txt](https://github.com/dronecore/DroneCore/blob/{{ book.github_branch }}/example/transition_vtol_fixed_wing/CMakeLists.txt)
+
+```make
+cmake_minimum_required(VERSION 2.8.12)
+
+project(transition_vtol_fixed_wing)
+
+if(NOT MSVC)
+    add_definitions("-std=c++11 -Wall -Wextra -Werror")
+else()
+    add_definitions("-std=c++11 -WX -W2")
+endif()
+
+add_executable(transition_vtol_fixed_wing
+    transition_vtol_fixed_wing.cpp
+)
+
+target_link_libraries(transition_vtol_fixed_wing
+    dronecore
+)
+```
+
+[fly_mission.cpp](https://github.com/dronecore/DroneCore/blob/{{ book.github_branch }}/example/transition_vtol_fixed_wing/transition_vtol_fixed_wing.cpp)
+
+```cpp
```

Comment:
At some point it would be nice to be able to reference and pull in source code from github, otherwise it might get outdated.

Reply:
Yes it would. There is a plugin that does this, but unfortunately it requires a version of node that is higher than is available on gitbook hosting. They tried to update node for me but that broke some other stuff so they had to revert. Some options:
1. I re-write the plugin to remove the ES6 promisy stuff.
2. We wait until their fancy pure-hosted solution comes out that sorts this out (but which is currently running lots behind schedule, and I am not yet sure will do everything we need)
3. I write some sort of clever auto-update script. 

I expect either 1 or 2 will happen, but not as high priority to pursue as other things on my list.

Reply:
Ah interesting, alright no worries.